### PR TITLE
Add fix suffix by mimetype to file processing

### PR DESCRIPTION
**Pull request by jonaskaz**
- adds python-magic dependency, the python module mimetype only includes a method that guesses mimetype by file suffix which would not work for this application

### CODE REVIEW

**Comment by Scott771** [comment]
https://github.com/ahupp/python-magic

Doesn't this require OS specific installations?

**Comment by Scott771** [comment]
if those dependencies aren't installed, how is the error handling?

File: indico_toolkit/pipelines/file_processing.py [hunk 1]
```diff
@@ -44,6 +47,39 @@ def get_file_paths_from_dir(
             f"Found {len(self.file_paths)} valid files and {len(self.invalid_suffix_paths)} paths with invalid suffixes."
         )
 
+    def fix_file_suffixes(
```
Comment (Scott771) [inline indico_toolkit/pipelines/file_processing.py:50]: is this renaming the files in place on the system??

Comment (Scott771) [reply indico_toolkit/pipelines/file_processing.py:50]: think that should be clearer or optional

Comment (Scott771) [reply indico_toolkit/pipelines/file_processing.py:50]: I would also probably be more opinionated with how the function works, i.e. I don' think we need to give the option to pass a mime_suffix_mapping (I would have no idea what to do with this)

File: setup.py [hunk 2]
```diff
@@ -18,6 +18,7 @@
         "setuptools>=41.4.0",
         "pandas>=1.0.3",
         "coverage>=5.5",
-        "PyMuPDF==1.18.13"
+        "PyMuPDF==1.18.13",
+        "python-magic>=0.4.24"
```
Comment (Scott771) [inline setup.py:22]: If you are using new pip installed libraries, make sure they are added to the setup.py and pyproject.toml (only pyproject.toml actually determines what gets installed as dependencies via pip)

**Review by Scott771 — CHANGES_REQUESTED** [review]
check python-magic requirements for different OS, if OK to use, then also add to .toml

**Comment by jonaskaz** [comment]
python-magic has os specific installation, will leave this feature as a standalone script

**Pull request closed without merging**